packages/offix-client/src/config/OffixClientConfig.ts

```diff
@@ -1,99 +1,55 @@
+import { createDefaultOfflineStorage } from "offix-offline";
+import { createDefaultCacheStorage } from "../cache";
 import { ApolloLink } from "apollo-link";
+import { CacheUpdates } from "offix-cache";
 import { ApolloOfflineQueueListener } from "../apollo";
 import { InMemoryCache } from "apollo-cache-inmemory";
 
 /**
- * Contains all configuration options required to initialize Voyager Client
- * Options marked with [Modifier] flag are used to modify behavior of client.
- * SDK provides default values for all [Modifier] flags.
- * Users do not need to pass them for normal initialization of the client.
- * Please refer to documentation for more information about the individual flag and it's side effects.
- *
+ * Class for managing user and default configuration.
+ * Default config is applied on top of user provided configuration
  */
-export interface OffixClientConfig {
-
-  /**
-   * The URL of http server that will be used to initialize default http link
-   * Value is ignored if terminating link is passed
-   */
-  httpUrl?: string;
-
-  /**
-   * [Modifier]
-   * Apollo link that will be passed to created client
-   */
-  terminatingLink?: ApolloLink;
-
-  /**
-   * [Modifier]
-   *
-   * The storage you want your client to use (Uses window.localStorage by default)
-   */
-  storage?: PersistentStore<PersistedData>;
-
-  /**
-   * [Modifier]
-   *
-   * The Apollo InMemoryCache you want your client to use. (Uses a default one if none provided)
-   */
-  cache?: InMemoryCache;
-
-  /**
-   * [Modifier]
-   *
-   * Interface for detecting changes in network status.
-   * See `WebNetworkStatus` and `CordovaNetworkStatus`
-   */
-  networkStatus?: NetworkStatus;
-
-  /**
-   * User provided listener that contains set of methods that can be used to detect
-   * when operations were added to queue
-   */
-  offlineQueueListener?: ApolloOfflineQueueListener;
-
-  /**
-   * [Modifier]
-   *
-   * Interface that defines how object state is progressed
-   * This interface needs to match state provider supplied on server.
-   */
-  conflictProvider?: ObjectState;
-
-  /**
-   * Interface that can be implemented to receive information about the data conflict
-   *
-   * @deprecated see OfflineClient.registerOfflineEventListener
-   */
-  conflictListener?: ConflictListener;
-
-  /**
-   * [Modifier]
-   *
-   * The conflict resolution strategy your client should use. By default it takes client version.
-   */
-  conflictStrategy?: ConflictResolutionStrategy;
-
-  /**
-   * [Modifier]
-   *
-   * Cache updates functions for your mutations
-   * Argument allows to restore optimistic responses on application restarts.
-   */
-  mutationCacheUpdates?: CacheUpdates;
-
-  /**
-   * [Modifier]
-   *
-   * The options to configure how failed offline mutations are retried.
-   *
-   */
-  retryOptions?: RetryLink.Options;
+export class OffixClientConfig implements OffixClientOptions {
+  public httpUrl?: string;
```

Comment:
That is breaking change. Any reason why we have done that :) ?

Reply:
This looks like a breaking change but it's actually not. We used to have two files. 

* OffixClientConfig - the interface that defined what options can be passed into the OfflineClient constructor by a user.
* OffixDefaultConfig - actual config class that we instantiate passing in user provided options and merging some defaults

I felt that the names were always really confusing and made little sense. So now we have the following:

* OffixClientOptions - Options represents the things a user can pass in
* OffixClientConfig - Config represents an actual instance of the config.

Because I renamed the files, it looks like I changed the options users can pass in but you'll actually find them all in `OffixClientOptions.ts` and there have been no changes at all.

Reply:
Perfect!